org-roam-node.el

```diff
@@ -476,6 +476,52 @@ The TEMPLATES, if provided, override the list of capture templates (see
        :templates templates
        :props '(:finalize find-file)))))
 
+;;;###autoload
+(defun org-roam-node-rename ()
+  "Prompts for new name for current node. Renames the node minding
+title/filename/link titles."
+  (interactive)
+  (unless (org-roam-buffer-p) (error "Not in an org-roam buffer."))
+  (save-some-buffers t)
+  (let* ((id (org-entry-get (point) "ID"))
+	 (node (org-roam-node-from-id id))
+	 (title (org-roam-get-keyword "title"))
+	 (newtitle (read-string "New title: " title)))
+    (when (null id) (error "Failed to fetch ID"))
+    (when (org-roam-node-from-title-or-alias newtitle)
+      (error (format "Node %s already exists" newtitle)))
+
+    ;; Set new title
+    (org-roam-set-keyword "title" newtitle) 
```

Comment:
This function only works for file nodes: it needs to generalize to headline nodes as well.